pkg/nsx/services/subnet/subnet.go

```diff
@@ -180,6 +180,30 @@ func (service *SubnetService) DeleteSubnet(nsxSubnet model.VpcSubnet) error {
 	return nil
 }
 
+func (service *SubnetService) ListSubnet() []model.VpcSubnet {
+	subnets := service.SubnetStore.List()
+	subnetSet := []model.VpcSubnet{}
+	for _, subnet := range subnets {
+		subnetSet = append(subnetSet, subnet.(model.VpcSubnet))
+	}
+	return subnetSet
+}
+
+// check if subnet belongs to a subnetset, if yes, check if that subnetset still exists
+func (service *SubnetService) IsOrphanSubnet(subnet model.VpcSubnet, subnetsets *v1alpha1.SubnetSetList) bool {
+	for _, tag := range subnet.Tags {
+		if *tag.Scope == common.TagScopeSubnetSetCRId {
+			subnetset_name := *tag.Tag
+			for _, subnetset := range subnetsets.Items {
```

Comment:
I am worried about the performance for the code.
Could we find another effecient way to check if the nsx subnet's reference SubnetSet CR exists or not?

Reply:
if we had a subnetset Store, the performance will be better.